src/test/java/com/google/TestCourseList.java

```diff
@@ -0,0 +1,38 @@
+package com.google;
+
+import static org.mockito.Mockito.*;
+
+import com.google.servlets.CourseListServlet;
+import java.io.*;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests CourseListServlet */
+@RunWith(JUnit4.class)
+public final class TestCourseList {
+  @Test
+  public void servletResponseHasCourses() {
+    HttpServletResponse mockedResponse = mock(HttpServletResponse.class);
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    try {
```

Comment:
is the try/catch necessary? Can it just fail the test when it throws an exception naturally?

Reply:
if I don't try/catch, it doesn't let me run the tests

Reply:
Try adding `throws Exception` to the function signature.